packages/platform-browser/test/IndexedDbDatabase.test.ts

```diff
@@ -0,0 +1,229 @@
+import {
+  IndexedDb,
+  IndexedDbDatabase,
+  IndexedDbMigration,
+  IndexedDbTable,
+  IndexedDbVersion
+} from "@effect/platform-browser"
+import { afterEach, assert, describe, it } from "@effect/vitest"
+import { Effect, Layer, Schema } from "effect"
+import { IDBKeyRange, indexedDB } from "fake-indexeddb"
+
+const databaseName = "db"
+
+const layerFakeIndexedDb = Layer.succeed(IndexedDb.IndexedDb, IndexedDb.make({ indexedDB, IDBKeyRange }))
+
+const provideMigration = (migration: IndexedDbMigration.IndexedDbMigration.Any) =>
+  Effect.provide(
+    IndexedDbDatabase.layer(databaseName, migration).pipe(Layer.provide(layerFakeIndexedDb))
+  )
+
+afterEach(() => {
+  indexedDB.deleteDatabase(databaseName)
+})
+
+describe.sequential("IndexedDbDatabase", () => {
+  it.effect("insert and read todos", () => {
+    const Table = IndexedDbTable.make(
+      "todo",
+      Schema.Struct({
+        id: Schema.Number,
+        title: Schema.String,
+        completed: Schema.Boolean
+      }),
+      { keyPath: "id", indexes: { titleIndex: "title" } }
+    )
+
+    const Db = IndexedDbVersion.make(Table)
+
+    class Migration extends IndexedDbMigration.make(Db, (api) =>
```

Comment:
It feels like `IndexedDbMigration` should actually be `IndexedDbDatabase`, and the `IndexedDbDatabase` tag can become more of an implementation detail.

Then `IndexedDbDatabase` could have a `getQueryBuilder` api for getting a query builder for the latest version.

Reply:
It works, removed `IndexedDbMigration` and unified all inside `IndexedDbDatabase`. A migration is created from `IndexedDbDatabase` and it has both a `getQueryBuilder` and `layer` api